src/features/deposit/components/DepositUIMachineProvider.tsx

```diff
@@ -239,6 +241,26 @@ export function DepositUIMachineProvider({
                   "chainType should be EVM"
                 )
 
+                if (tokenAddress !== "native") {
+                  const allowance = await getAllowance(
+                    tokenAddress,
+                    accountId,
+                    siloToSiloAddress.turbochain,
+                    BlockchainEnum.TURBOCHAIN
+                  )
+                  assert(allowance != null, "Allowance is not defined")
```

Comment:
You'll log the same error twice when allowance request fails: in the caller of this function and getAllowance itself.